app/models/smart_proxy.rb

```diff
@@ -88,12 +86,19 @@ def has_feature?(feature)
     self.features.any? { |proxy_feature| proxy_feature.name == feature }
   end
 
-  private
-
   def proxy_status
-    @proxy_status ||= ProxyStatus.new(self)
+    @proxy_status ||= { :version => ProxyStatus::Version.new(self) }
+    features.map(&:name).each do |feature|
+      proxy_status_class = "ProxyStatus::#{feature.delete(' ')}".constantize rescue nil
```

Comment:
the reason was that STI wouldn't work correctly but that seems to be right solution in here as well